Run cbmc c++ regression as part of default regression test

andreast271 opened the pull request:
Running cbmc on c++ files was not part of standard regression tests, which lead to a regression. 
This change set fixes the regression, adds a few tests, and adds cbmc c++ tests to the standard regression.

andreast271 committed:
Run cbmc c++ regression as part of default regression test
Set is_parameter for c++ function parameter symbol

reuk reviewed (commented):
Should be enabled in the cmake files too.

andreast271 commented:
There are other test directories in the Makefile that are missing from cmake: cbmc-cover, goto-analyzer-taint, goto-gcc. Should they also be added?

reuk commented:
In that case, don't worry about it. I'll put together a PR enabling the remaining directories once this has been merged.

andreast271 commented:
I needed to add a commit to this PR anyway to disable the new tests on appveyor. I added the missing test directories to cmake at the same time. Exception: goto-gcc. I wasn't sure about the correct cmake file for it.

andreast271 committed:
Add tests to cmake regression: cbmc-cover, cbmc-cpp, goto-analyzer-taint

andreast271 committed:
Disable cbmc-cpp tests in appveyor, which runs regression on windows.
All cbmc-cpp tests #include <assert.h> and cbmc cannot yet parse
Microsoft C++ headers.

thk123 commented:
I think it is not worth at this stage getting these tests to work on Windows (they would need to be wrapped in some kind of script to do the fixing up). Perhaps something like this for now?

```
if(NOT WIN32)
    add_test_pl_tests(
      "$<TARGET_FILE:goto-cc>"
    )
endif()
```

@reuk can say whether this is a good way of doing this.

andreast271 commented:
Some of the goto-gcc tests fail on Travis when running as part of the cmake build. See https://travis-ci.org/diffblue/cbmc/jobs/289820097. 
I will revert the commit that added the goto-gcc tests to the cmake build. Fixing goto-gcc tests under cmake is not the point of the PR.

thk123 commented:
Yup that's reasonable - I've raised an issue #1500 to remind us to fix this, but it certainly shouldn't block this PR.

tautschnig commented on appveyor.yml:
```diff
@@ -83,6 +83,7 @@ test_script:
     rmdir /s /q cbmc\byte_update7
     rmdir /s /q cbmc\pipe1
     rmdir /s /q cbmc\unsigned___int128
+    rmdir /s /q cbmc-cpp
```
Would this work if we had #1260 merged?

andreast271 replied:
Yes. You should be able to merge this PR and #1260 in any order. I'd recommend merging them close to each other.

tautschnig replied:
How about merging this one first, and then reverting this particular change in #1260?

andreast271 replied:
Sorry, I misunderstood. The cbmc-cpp tests never ran on any platform. This PR enables them in general, and then disables them specifically for windows. They don't work on windows irrespective of #1260, since they #include <assert.h>.
